cmds/insmod/insmod.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"io/ioutil"
+	"syscall"
+	"unsafe"
+)
+
+func main() {
+	var options string
+
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "insmod: ERROR: missing filename.\n")
+		os.Exit(1)
+	}
+
+	// get filename from argv[1]
+	filename := os.Args[1]
+
+	// Everything else is module options
+	for i := 2; i < len(os.Args); i++ {
+		options = options + os.Args[i] + " "
+	}
+
+	// read file into memory
+	file, err := ioutil.ReadFile(filename)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "insmod: can't read '%s': %v\n", filename, err)
```

Comment:
u-root likes log.Fatalf for stuff like this, since it might run in an unattended environment.